pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -366,6 +379,54 @@ class AnalyticsImpl implements Analytics {
   @override
   void close() => _gaClient.close();
 
+  @override
+  Future<List<Survey>> fetchAvailableSurveys() async {
+    final List<Survey> surveysToShow = [];
+    final LogFileStats? logFileStats = _logHandler.logFileStats();
+
+    for (final Survey survey in await _surveyHandler.fetchSurveyList()) {
+      // Counter to check each survey condition, if all are met, then
+      // this integer will be equal to the number of conditions in
+      // [Survey.conditionList]
+      int conditionsMet = 0;
+      for (final Condition condition in survey.conditionList) {
+        // Retrieve the value from the [LogFileStats] with
+        // the label provided in the condtion
+        final int? logFileStatsValue =
+            logFileStats?.getValueByString(condition.field);
+
+        if (logFileStatsValue == null) continue;
+
+        switch (condition.operatorString) {
+          case '>=':
+            if (logFileStatsValue >= condition.value) conditionsMet++;
+            break;
+          case '<=':
+            if (logFileStatsValue <= condition.value) conditionsMet++;
+            break;
+          case '>':
+            if (logFileStatsValue > condition.value) conditionsMet++;
+            break;
+          case '<':
+            if (logFileStatsValue < condition.value) conditionsMet++;
+            break;
+          case '==':
+            if (logFileStatsValue == condition.value) conditionsMet++;
+            break;
+          case '!=':
+            if (logFileStatsValue != condition.value) conditionsMet++;
+            break;
+        }
```

Comment:
what happens if we fall through and don't hit any of these? I'm guessing we should have a default that throws?

Reply:
If for some reason the operator in the metadata file wasn't accounted for in this switch statement, then the survey would not be delivered to the end user, which i think is preferable over an error getting surfaced?

But like you mentioned in your other comment, we will look into creating something that properly vets each time we upload metadata files so that `package:unified_analytics` is able to parse them